include/ast.hpp

```diff
@@ -194,6 +194,69 @@ struct IntConstExprNode : public ExprNode {
   int val;
 };
 
+/// @note This is an abstract class.
+struct UnaryExprNode : public ExprNode {
+  UnaryExprNode(std::unique_ptr<ExprNode> clause) : clause{std::move(clause)} {}
+
+  virtual void Accept(NonModifyingVisitor&) const override;
+  virtual void Accept(ModifyingVisitor&) override;
+
+  /// @note To make the class abstract.
+  virtual ~UnaryExprNode() = 0;
+
+  std::unique_ptr<ExprNode> clause;
+};
+
+struct IncrExprNode : public UnaryExprNode {
+  using UnaryExprNode::UnaryExprNode;
+
+  virtual void Accept(NonModifyingVisitor&) const override;
+  virtual void Accept(ModifyingVisitor&) override;
+};
+
+struct DecrExprNode : public UnaryExprNode {
+  using UnaryExprNode::UnaryExprNode;
+
+  virtual void Accept(NonModifyingVisitor&) const override;
+  virtual void Accept(ModifyingVisitor&) override;
+};
+
+struct NegExprNode : public UnaryExprNode {
+  using UnaryExprNode::UnaryExprNode;
+
+  virtual void Accept(NonModifyingVisitor&) const override;
+  virtual void Accept(ModifyingVisitor&) override;
+};
+
+struct AddrExprNode : public UnaryExprNode {
+  using UnaryExprNode::UnaryExprNode;
+
+  virtual void Accept(NonModifyingVisitor&) const override;
+  virtual void Accept(ModifyingVisitor&) override;
+};
+
+struct DereferExprNode : public UnaryExprNode {
+  using UnaryExprNode::UnaryExprNode;
+
+  virtual void Accept(NonModifyingVisitor&) const override;
+  virtual void Accept(ModifyingVisitor&) override;
+};
+
+struct NotExprNode : public UnaryExprNode {
+  using UnaryExprNode::UnaryExprNode;
+
+  virtual void Accept(NonModifyingVisitor&) const override;
+  virtual void Accept(ModifyingVisitor&) override;
+};
+
+/// @note This is a bitwise complement unary expression node.
```

Comment:
```suggestion
/// @brief This is a bitwise complement unary expression node.
```
This is a concise description rather than highlighting specific properties, so we use `@brief`.